### ISSUE TITLE
[v0.7.0] Random "No stored state - unable to handle response" messages

### ISSUE THREAD

**Issue by tomasIHS** [openid/AppAuth-Android#273]
Hi,
  
 I'm finding a lot of times that I receive the "No stored state - unable to handle response" message. After inspecting AppAuth source code, this happens after coming back from the browser and savedInstanceState on the onCreate is null.

 I put some logs and the onSaveInstance is always called, but anyway, sometimes savedInstanceState is still null. This should only happen if the OS destroys the activity entirely which it's odd. It happens more with debug versions of my app than with release versions.

 I guess a solution would be serializing to disk instead of using savedInstanceState, but it would be complicated due to the pending intents. At least, it would be nice if it's possible to pass an activity class, so AppAuth calls it in case a "No stored state - unable to handle response" message happens.

Any clue about what I'm doing wrong to trigger this behaviour? Thanks.

**Comment by tomasIHS** [comment]
Ok, the first time it logins correctly, but after logging out and repeating the operation (login), it doesn't:

Output of dumpsys the first time when the browser with the oAuth login is shown:
``
  Stack #1:
    Running activities (most recent first):
      TaskRecord{aec6e54 #15078 A=org.mozilla.focus U=0 sz=1}
        Run #1: ActivityRecord{bf35bc7 u0 org.mozilla.focus/.activity.MainActivity t15078}
      TaskRecord{cca19cd #15077 A=com.tomasihs.myfineapp U=0 sz=1}
        Run #0: ActivityRecord{327733a u0 com.tomasihs.myfineapp/net.openid.appauth.AuthorizationManagementActivity t15077}
``

Output of dumpsys the second time:
``
  Stack #1:
    Running activities (most recent first):
      TaskRecord{aec6e54 #15078 A=org.mozilla.focus U=0 sz=1}
        Run #0: ActivityRecord{bf35bc7 u0 org.mozilla.focus/.activity.MainActivity t15078}
``

So basically, when the user hits the browser the second time, my app is not running. I don't get why.

**Comment by tomasIHS** [comment]
Definitely there was something wrong in my app. Not disposing the service was helping keeping it alive, but now I can dispose the service and works everytime. I'll close the issue.

**Comment by VigneshKSI** [comment]
Hi ,
I am getting error also like this 

"No stored state - unable to handle response" and in AuthorizationManagmentActivity oncreate is returning null..

Can anyone Please help to solve this,I was trying this more than a months..

**Comment by VigneshKSI** [comment]
I am using the pie version 9.0...

**Comment by vvavepacket** [comment]
Getting this on 10.0 as well

2020-08-15 11:01:12.051 19482-19545/com.xxxx V/FA: onActivityCreated
2020-08-15 11:01:12.078 19482-19482/com.xxxx W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@637841f
2020-08-15 11:01:12.081 19482-19482/com.xxxx W/AppAuth: No stored state - unable to handle response

**Comment by dIeGoLi** [comment]
The problem is not with the library. I want to describe my cause of the problem since it maybe gives hints to others.
Everything worked fine, until i merged the master branch of my company app into my feature branch :).

The flow is intended to work like that:
1. Your Login Activity/Fragment calls when ready `startActivityForResult(authIntent, RC_AUTH);` . To trigger that the library opens the custom tab etc. You expect the result of the process to be delivered to onActivityResult.
2. The intent used in (1) opens AuthorizationManagementActivity (which is launchMode singleTask). It saves the relevant "state" in on save instance state.
3. The Browser Intent is executed. When successfull the callback is executed
4. RedirectUriReceiverActivity catches the callback. Starts AuthorizationManagementActivity. Which will be the same instance than in (2)
5. AuthorizationManagementActivity  reads the previously saved state and sets the result so your onActivityResult is being called.

In my case somebody registered in the master branch for the application a ActivityLifecycleCallbacks which basically destroys every activity which is resumed while not in the "logged in" state, by redirecting to the initial Login activity with  `Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK`
So the observable behaviour in my case was that my Login Activity got destroyed after calling `startActivityForResult`. But the real problem is, that AuthorizationManagementActivity got finished as well (clear task), which means in Step 4 a new instance of AuthorizationManagementActivity is created which will have "no stored state". Hope it helps to others ...

**Comment by denys-meloshyn** [comment]
In my case I used `Rx` to observe when to start OAuth flow and I didn't dispose it correctly, as result I started login twice.
